numalogic/udfs/inference.py

```diff
@@ -144,17 +136,10 @@ def exec(self, keys: list[str], datum: Datum) -> Messages:
 
         # Send training request if artifact loading is not successful
         if not artifact_data:
-            payload = replace(
-                payload, status=Status.ARTIFACT_NOT_FOUND, header=Header.TRAIN_REQUEST
-            )
-            _increment_counter(
-                counter=MODEL_STATUS_COUNTER,
-                labels=(
-                    payload.status.value,
-                    *_metric_label_values,
-                ),
-            )
-            return Messages(Message(keys=keys, value=payload.to_json()))
+            msgs = Messages(get_trainer_message(keys, _stream_conf, payload))
```

Comment:
would be good if we also keep the artifact status along side sending trainer message here.

Reply:
The trainer payload does not really have a status field.